src/components/root/createModals/CreateScriptModal.tsx

```diff
@@ -0,0 +1,99 @@
+import {
+	Button,
+	DialogActions,
+	DialogContent,
+	DialogTitle,
+	Divider,
+	FormControl,
+	FormHelperText,
+	FormLabel,
+	Input,
+	Modal,
+	ModalDialog,
+} from '@mui/joy';
+import { CreateModalsProps } from './createModalsProps';
+import { iconFromTabType, Script } from '../../../types/application-data/application-data';
+import { useEffect, useMemo, useState } from 'react';
+import { toValidFunctionName } from '../../../utils/string';
+import { InfoOutlined } from '@mui/icons-material';
+import { useAppDispatch } from '../../../state/store';
+import { addTabs, setSelectedTab } from '../../../state/tabs/slice';
+import { createScript } from '../../../state/active/thunks/scripts';
+
+export function CreateScriptModal({ open, closeFunc }: CreateModalsProps) {
+	const [scriptName, setScriptName] = useState('');
+	const [scriptCallingName, setScriptCallingName] = useState('');
+	const dispatch = useAppDispatch();
+	const createScriptFunction = async () => {
+		const newScript: Partial<Script> = { name: scriptName, scriptCallableName: scriptCallingName, content: '' };
+
+		const createdScriptId = await dispatch(createScript(newScript)).unwrap();
+		dispatch(addTabs({ [createdScriptId]: 'script' }));
+		dispatch(setSelectedTab(createdScriptId));
+	};
+	const scriptCallingNameValid =
+		scriptCallingName.length > 0 && toValidFunctionName(scriptCallingName) === scriptCallingName;
+	const scriptNameValid = scriptName.length > 0;
+	const allFieldsValid = useMemo(() => scriptNameValid && scriptCallingNameValid, [scriptName, scriptCallingName]);
+
+	useEffect(() => {
```

Comment:
I don't think useEffects are useful here. You can just define a `const scriptCallingName = toValidFunctionName(scriptName)` and avoid the re-render that'll happen by using a useEffect to set it. In the end it's more performant and easier to follow imo.

Reply:
but scriptName is state? useEffect is the best way to listen for state change

Reply:
Ack, nvm. I see you're setting it elsewhere. Auto-resolving.

Reply:
I want it to update every time scriptName changes

Reply:
It will if you just have it as a const. I don't think useEffect is the best way to listen for state changes, but this is probably a larger conversation. I think useEffect is valid here because you allow both to be updated in state so you need some way to synchronize them.